src/PointwiseFunctions/GeneralRelativity/ProjectionOperators.cpp

```diff
@@ -0,0 +1,155 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/ProjectionOperators.hpp"
+
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// \cond
+namespace gr {
+template <size_t VolumeDim, typename Frame, typename DataType>
+tnsr::II<DataType, VolumeDim, Frame> transverse_projection_operator(
+    const tnsr::II<DataType, VolumeDim, Frame>& inverse_spatial_metric,
+    const tnsr::I<DataType, VolumeDim, Frame>& normal_vector) noexcept {
+  tnsr::II<DataType, VolumeDim, Frame> projection_tensor{};
+  transverse_projection_operator(make_not_null(&projection_tensor),
+                                 inverse_spatial_metric, normal_vector);
+  return projection_tensor;
+}
+
+template <size_t VolumeDim, typename Frame, typename DataType>
+void transverse_projection_operator(
+    const gsl::not_null<tnsr::II<DataType, VolumeDim, Frame>*>
+        projection_tensor,
+    const tnsr::II<DataType, VolumeDim, Frame>& inverse_spatial_metric,
+    const tnsr::I<DataType, VolumeDim, Frame>& normal_vector) noexcept {
+  if (UNLIKELY(get_size(get<0, 0>(*projection_tensor)) !=
```

Comment:
include `Utilities/ContainerHelpers.hpp` and use `destructive_resize_components(projection_tensor, get<0>(normal_vector).size())` If `destructive_resize_components` doesn't work on `double`, then just wrap in an `if constexpr (not std::is_same_v<double, DataType>)`

Reply:
done.